src/BaseSlackHandler.ts

```diff
@@ -0,0 +1,207 @@
+import { Logging } from "matrix-appservice-bridge";
+import * as rp from "request-promise-native";
+
+import { Main } from "./Main";
+import { SlackGhost } from "./SlackGhost";
+import { default as subs } from "./substitutions";
+
+const log = Logging.get("BaseSlackHandler");
+
+const CHANNEL_ID_REGEX = /<#(\w+)\|?\w*?>/g;
+const CHANNEL_ID_REGEX_FIRST = /<#(\w+)\|?\w*?>/;
+
+// (if the message is an emote, the format is <@ID|nick>, but in normal msgs it's just <@ID>
+const USER_ID_REGEX = /<@(\w+)\|?\w*?>/g;
+const USER_ID_REGEX_FIRST = /<@(\w+)\|?\w*?>/;
+
+export const INTERNAL_ID_LEN = 32;
+export const HTTP_CODES = {
+    CLIENT_ERROR: 400,
+    FORBIDDEN: 403,
+    NOT_FOUND: 404,
+    OK: 200,
+    SERVER_ERROR: 500,
+};
+
+export interface ISlackMessage {
+    channel: string;
+    text?: string;
+}
+
+export interface ISlackFile {
+    permalink_public: string;
+    id: string;
+    url_private: string;
+    public_url_shared: string;
+// tslint:disable-next-line: no-any
+    _content: any;
```

Comment:
Private fields should not be on public interfaces

Reply:
Reminder to myself to get back to this one, it's used all over the place.